### DISCUSSION TITLE
Electron app shows empty window when exe installer is double clicked

### DISCUSSION THREAD

**Discussion by praveen69sk** [eclipse-theia/theia#11319] [Q&A]
I'm trying to build a simple Hello-World electron application,
-  it works fine when I launch it from from the terminal using 'yarn start' from electron-app package.json
-  it fails when I create exe installer for windows and launch the app  it shows empty window

Environment:
    Node v12.22.0
    npm@6.14.11
    Windows 10

package structure-
**Hello-World-App
    |-browser-app
    |-electron-app
    |-package.json**

<details>

### root package.json content
> {
  "private": true,
  "scripts": {
    "prepare": "lerna run prepare",
    "rebuild:browser": "theia rebuild:browser",
    "rebuild:electron": "set THEIA_ELECTRON_SKIP_REPLACE_FFMPEG=1 && theia rebuild:electron",
    "start:browser": "yarn rebuild:browser && yarn --cwd browser-app start",
    "start:electron": "yarn --cwd electron-app start",
    "electron": "yarn --cwd electron-app",
    "watch": "lerna run --parallel watch"
  },
  "devDependencies": {
    "lerna": "2.4.0"
  },
  "resolutions": {
    "**/@theia/application-manager": "1.20.0",
    "**/@theia/application-package": "1.20.0",
    "**/@theia/callhierarchy": "1.20.0",
    "**/@theia/cli": "1.20.0",
    "**/@theia/console": "1.20.0",
    "**/@theia/core": "1.20.0",
    "**/@theia/debug": "1.20.0",
    "**/@theia/editor": "1.20.0",
    "**/@theia/electron": "1.20.0",
    "**/@theia/file-search": "1.20.0",
    "**/@theia/filesystem": "1.20.0",
    "**/@theia/git": "1.20.0",
    "**/@theia/keymaps": "1.20.0",
    "**/@theia/markers": "1.20.0",
    "**/@theia/messages": "1.20.0",
    "**/@theia/mini-browser": "1.20.0",
    "**/@theia/monaco": "1.20.0",
    "**/@theia/navigator": "1.20.0",
    "**/@theia/outline-view": "1.20.0",
    "**/@theia/output": "1.20.0",
    "**/@theia/plugin-dev": "1.20.0",
    "**/@theia/plugin-ext": "1.20.0",
    "**/@theia/plugin-ext-vscode": "1.20.0",
    "**/@theia/plugin-metrics": "1.20.0",
    "**/@theia/preferences": "1.20.0",
    "**/@theia/process": "1.20.0",
    "**/@theia/scm": "1.20.0",
    "**/@theia/scm-extra": "1.20.0",
    "**/@theia/search-in-workspace": "1.20.0",
    "**/@theia/task": "1.20.0",
    "**/@theia/terminal": "1.20.0",
    "**/@theia/typehierarchy": "1.20.0",
    "**/@theia/userstorage": "1.20.0",
    "**/@theia/variable-resolver": "1.20.0",
    "**/@theia/vsx-registry": "1.20.0",
    "**/@theia/workspace": "1.20.0"
  },
  "workspaces": [
    "hello-world", "browser-app", "electron-app"
  ]
}

### electron-app package.json content

> {
  "private": true,
  "name": "electron-app",
  "main":"src-gen/frontend/electron-main.js",
  "author": "Praveen S Kalawad",
  "homepage": "https://github.com/eclipse-theia/theia-blueprint#readme",
  "version": "0.0.0",
  "repository": {
    "type": "git",
    "url": "git+https://github.com/eclipse-theia/theia-blueprint.git"
  },
  "engines": {
    "yarn": ">=1.7.0 <2",
    "node": ">=12.14.1"
  },
  "dependencies": {
    "@theia/core": "^1.0.0",
    "@theia/editor": "^1.0.0",
    "@theia/electron": "^1.0.0",
    "@theia/filesystem": "^1.0.0",
    "@theia/markers": "^1.0.0",
    "@theia/messages": "^1.0.0",
    "@theia/monaco": "^1.0.0",
    "@theia/navigator": "^1.0.0",
    "@theia/preferences": "^1.0.0",
    "@theia/process": "^1.0.0",
    "@theia/terminal": "^1.0.0",
    "@theia/workspace": "^1.0.0",
    "hello-world": "0.0.0"
  },
  "devDependencies": {
    "@theia/cli": "^1.0.0",
    "electron": "15.3.5",
    "electron-builder": "22.10.5",
    "app-builder-lib": "22.10.5"
  },
  "scripts": {
    "prepare": "yarn build",
    "clean": "theia clean && rimraf node_modules",
    "clean:dist": "rimraf dist",
    "build": "yarn bundle",
    "bundle": "theia build --mode development",
    "watch": "concurrently -n compile,bundle \"theiaext watch --preserveWatchOutput\" \"theia build --watch --mode development\"",
    "start": "theia start",
    "start:debug": "yarn start --log-level=debug",
    "package": "yarn clean:dist  && electron-builder -c.mac.identity=null --publish never",
    "deploy": "yarn clean:dist && electron-builder -c.mac.identity=null --publish always",
    "package:preview": "yarn clean:dist && yarn rebuild && yarn build && electron-builder -c.mac.identity=null --dir",
    "update:checksum": "ts-node scripts/update-checksum.ts",
    "update:next": "ts-node ../../scripts/update-theia-to-next.ts",
    "download:plugins": "theia download:plugins",
    "test": "mocha --timeout 60000 \"./test/*.spec.js\"",
    "rebuild": "theia rebuild:electron"
  },
  "theia": {
    "target": "electron",
    "frontend": {
      "config": {
        "applicationName": "Hello World App",
        "preferences": {
          "toolbar.showToolbar": true
        }
      }
    }
  }
}

</details>

I would be grateful for any help.

**Comment by msujew** [comment]
Hi @praveen69sk,

There are likely some issues logged in the frontend console (<kbd>Ctrl+Shift+I</kbd>) even if the app doesn't start correctly. There might also be something in the backend log. Can you share these errors, if there are any? Without a repo to reproduce any issues or error logs, there's little we can help you with. The package.json configuration looks fine, but that's rarely an indicator for potential issues.

**Reply by praveen69sk** [reply]
Hi @msujew,

 Thanks for the quick response. 
- The App works fine for both browser-app & electron-app when i run  **yarn -- > theia start** 
- But fails when I am creating exe and running it via command line, it's throwing the following error

**Please find the error info:**

```
 C:\PraveenSK\Work\Learning\Theia-Windows\Electron-Examples\hello-world-app_1_20_electron_workingcopy\electron-app\dist\win-unpacked>start Hello-World-App.exe

C:\PraveenSK\Work\Learning\Theia-Windows\Electron-Examples\hello-world-app_1_20_electron_workingcopy\electron-app\dist\win-unpacked>
C:\PraveenSK\Work\Learning\Theia-Windows\Electron-Examples\hello-world-app_1_20_electron_workingcopy\electron-app\dist\win-unpacked\resources\app\node_modules\bindings\bindings.js:121
        throw e;
        ^

Error: Module did not self-register: '\\?\C:\PraveenSK\Work\Learning\Theia-Windows\Electron-Examples\hello-world-app_1_20_electron_workingcopy\electron-app\dist\win-unpacked\resources\app\node_modules\drivelist\build\Release\drivelist.node'.
    at process.func [as dlopen] (electron/js2c/asar.js:140:31)
    at Object.Module._extensions..node (internal/modules/cjs/loader.js:1034:18)
    at Object.func [as .node] (electron/js2c/asar.js:140:31)
    at Module.load (internal/modules/cjs/loader.js:815:32)
    at Module._load (internal/modules/cjs/loader.js:727:14)
    at Function.Module._load (electron/js2c/asar.js:769:28)
    at Module.require (internal/modules/cjs/loader.js:852:19)
    at require (internal/modules/cjs/helpers.js:74:18)
    at bindings (C:\PraveenSK\Work\Learning\Theia-Windows\Electron-Examples\hello-world-app_1_20_electron_workingcopy\electron-app\dist\win-unpacked\resources\app\node_modules\bindings\bindings.js:112:48)
    at Object.<anonymous> (C:\PraveenSK\Work\Learning\Theia-Windows\Electron-Examples\hello-world-app_1_20_electron_workingcopy\electron-app\dist\win-unpacked\resources\app\node_modules\drivelist\js\index.js:25:27)
```

![image](https://user-images.githubusercontent.com/103233516/174782900-c94dc4f6-0a2d-4fa3-9205-cc031a2cbf7b.png)

**Reply by vince-fugnitto** [reply]
> But fails when I am creating exe and running it via command line, it's throwing the following error

@praveen69sk how are you creating an `exe`?
You can take a look at [theia-blueprint](https://github.com/eclipse-theia/theia-blueprint) as an example, it uses [electron-builder](https://www.electron.build/).

**Reply by praveen69sk** [reply]
Hi @msujew & @vince-fugnitto  ,

Yes I am referring [theia-blueprint](https://github.com/eclipse-theia/theia-blueprint) only, 
- Here I have used '**yarn package:preview'** then I got this launcher/exe 
- If I use **'yarn package'** it will give me installable exe, but after the installation in both the cases I am getting the same error.

Could you please help me with this.

**Reply by praveen69sk** [reply]
Hi @vince-fugnitto , @msujew 

After using electron version 15.3.5 and referring the [theia-blueprint](https://github.com/eclipse-theia/theia-blueprint) I am getting the following error.

Plz have a look at the screenshot

![image](https://user-images.githubusercontent.com/103233516/174844117-68ebebc1-2c3b-48b9-acf6-76dc41247e5b.png)

**Reply by praveen69sk** [reply]
Hi @vince-fugnitto , @msujew,

Could you please provide me the solution as it might be very helpful me to proceed further. I got stucked at this place now.

Thanks in advance.

**Reply by vince-fugnitto** [reply]
@praveen69sk unfortunately I do not have a windows machine to help further, when I built, packaged (preview and non-preview) the application worked fine for me (Linux). Are you performing any modifications to the codebase, or are you building the latest [master](https://github.com/eclipse-theia/theia-blueprint)?

**Reply by praveen69sk** [reply]
Hi @vince-fugnitto,

Thanks for the reply!

I am not building with the latest [master](https://github.com/eclipse-theia/theia-blueprint) I am using 1.20.0 version so I need to create Desktop app for this version only.

As you have mentioned it's working for you, Is that for latest master or for 1.20.0 version? Plz let me know

**Reply by vince-fugnitto** [reply]
I've been using `master` and on I'm Linux so it is quite different from your use-case.
If you still experience issues on `master`, you can open an issue directly on that repo for one of the authors to confirm (https://github.com/eclipse-theia/theia-blueprint/issues).

**Reply by praveen69sk** [reply]
Hi @vince-fugnitto,

Thanks for the quick response!

For me master is working fine, but our project is using theia 1.20.0 version so i need to create a exe or debian package for it.

As per your suggestion i have opened a new issue in https://github.com/eclipse-theia/theia-blueprint/issues/222
